site/app/libraries/database/DatabaseQueries.php

```diff
@@ -3257,15 +3257,14 @@ private function updateGradeableComponents(\app\models\gradeable\Gradeable $grad
      * @param \app\models\gradeable\Gradeable $gradeable The gradeable to update
      */
     public function updateGradeable(\app\models\gradeable\Gradeable $gradeable) {
-
         // If the gradeable has been modified, then update its properties
         if ($gradeable->isModified()) {
             $params = [
                 $gradeable->getTitle(),
                 $gradeable->getInstructionsUrl(),
                 $gradeable->getTaInstructions(),
                 $gradeable->getType(),
-                $this->course_db->convertBoolean($gradeable->isGradeByRegistration()),
+                $gradeable->getGraderAssignmentMethod(),
```

Comment:
There's a similar line to this in `CreateGradeable` that also needs to be changed (3132).  This may be what's causing your crash when creating a gradeable.

Reply:
Yes, I changed this and it seems to be working.  Thanks for the help!